src/control/server/mgmt_drpc.go

```diff
@@ -23,57 +23,59 @@
 
 package main
 
+// #cgo CFLAGS: -I../../include
+// #include <daos/drpc_modules.h>
+import "C"
 import (
-	"os"
-	"path/filepath"
+	"fmt"
 
+	pb "github.com/daos-stack/daos/src/control/common/proto/mgmt"
 	"github.com/daos-stack/daos/src/control/drpc"
+	"github.com/golang/protobuf/proto"
 	"github.com/pkg/errors"
 )
 
-var sockFileName = "daos_server.sock"
+const (
+	mgmtModuleID   int32 = C.DRPC_MODULE_MGMT
+	methodKillRank int32 = C.DRPC_METHOD_MGMT_KILL_RANK
+)
+
+// MgmtModule is the management drpc module struct
+type MgmtModule struct{}
+
+// InitModule is empty for this module
+func (m *MgmtModule) InitModule(state drpc.ModuleState) {}
 
-func getDrpcClientSocket(sockDir string) string {
-	return filepath.Join(sockDir, "daos_io_server.sock")
+// ID will return Mgmt module ID
+func (m *MgmtModule) ID() int32 {
+	return mgmtModuleID
 }
 
-func getDrpcClientConnection(sockDir string) *drpc.ClientConnection {
-	clientSock := getDrpcClientSocket(sockDir)
-	return drpc.NewClientConnection(clientSock)
+// MakeCall connects creates and sends message through the MgmtModule
+//func (m *MgmtModule) MakeCall(client *drpc.Client, method int32, body []byte) ([]byte, error) {
+
+// HandleCall is the handler for calls to the MgmtModule
+func (m *MgmtModule) HandleCall(client *drpc.Client, method int32, body []byte) ([]byte, error) {
+	return nil, errors.New("mgmt module handler is not implemented")
 }
 
-// drpcSetup creates socket directory, specifies socket path and then
-// starts drpc server.
-func drpcSetup(sockDir string) error {
-	// Create our socket directory if it doesn't exist
-	_, err := os.Stat(sockDir)
-	if err != nil && os.IsPermission(err) {
-		return errors.Wrap(
-			err, "user does not have permission to access "+sockDir)
-	} else if err != nil && os.IsNotExist(err) {
-		err = os.MkdirAll(sockDir, 0755)
-		if err != nil {
-			return errors.Wrap(
-				err,
-				"unable to create socket directory "+sockDir)
-		}
-	}
+// callDrpcMethodWithMessage create a new drpc Call instance, open a
+// drpc connection, send a message with the protobuf message marshalled
+// in the body, and closes the connection. Returns unmarshalled response.
+func (c *controlService) callDrpcMethodWithMessage(
+	methodID int32, body proto.Message) (resp *pb.DaosResponse, err error) {
 
-	sockPath := filepath.Join(sockDir, sockFileName)
-	drpcServer, err := drpc.NewDomainSocketServer(sockPath)
+	drpcResp, err := makeDrpcCall(c.drpc, mgmtModuleID, methodID, body)
 	if err != nil {
-		return errors.Wrap(err, "unable to create socket server")
+		fmt.Printf("%+v\n", err)
+		return nil, errors.WithStack(err)
 	}
 
-	// Create and add our modules
-	secmodule := &SecurityModule{}
-	drpcServer.RegisterRPCModule(secmodule)
-
-	err = drpcServer.Start()
+	resp = &pb.DaosResponse{}
+	err = proto.Unmarshal(drpcResp.Body, resp)
```

Comment:
If there is no body, this could fail. I've seen mixed results in cases like this - sometimes it produces an empty structure instead.
I'd probably move the processing of the body to the caller in mgmt_iosrv.go instead. Will every mgmt function return a DaosResponse?

Reply:
I imagine so